packages/web3-providers/src/utils/index.ts

```diff
@@ -0,0 +1,36 @@
+import { ChainId, createERC20Token, createNativeToken, formatBalance } from '@masknet/web3-shared-evm'
+import BigNumber from 'bignumber.js'
+
+export function getOrderUnitPrice(currentPrice?: string, decimals?: number, quantity?: string) {
+    if (!currentPrice || !decimals || !quantity) return
+    const price = formatBalance(currentPrice, decimals)
+    const _quantity = formatBalance(quantity, new BigNumber(quantity).toString() !== '1' ? 8 : 0)
+    return new BigNumber(price).dividedBy(_quantity).toFixed(4, 1).toString()
```

Comment:
FYI. https://github.com/DimensionDev/Maskbook/pull/5126/files

Reply:
why to string?